Add wp_content_owner and wp_content_group parameters

treydock opened the pull request:
Add wp_content_owner and wp_content_group parameters to allow setting different permissions on wp-content.  This is useful in an attempt to lock down ownership of Wordpress files but still allow themes and plugins to be installed from the web interface.

cfeskens commented:
Merging this would be a great help to us, as we often have to grant file permissions to the wp-content directory to allow web developers to modify themes.

treydock commented:
I've rebased this PR and also added the `wp_content_recurse` boolean option to disable recursive permission management on wp-content.  Default retains existing behavior.

freibuis commented:
is this PR going to happen?

I would sugguest adding 
wp_content_upload owner/group as well... this way uploads can be written to but themes/pluggins cannot ;)
